enhancements/installer/aws-permissions-check-bypass.md

```diff
@@ -0,0 +1,307 @@
+already be performed by the AWS permissions simulation API.
+
+## Proposal
+
+### Installer
+Introduce an install-config.yaml field that can be populated by users to
+indicate that the installer should not concern itself with determining whether
+the credentials provided are sufficient for installation and to affect the
+in-cluster behavior of cloud-credential-operator.
+
+Extend the install-config type in the installer repo:
+```
+type InstallConfig struct {
+	// ForceCredentialsMode instructs the installer to not attempt to query
+	// the cloud permissions before attempting installation. It also passes
+	// down the desired credentials mode to the cloud-credential-operator
+	// so that it too does not attempt to query permissions.
+	// +optional
+	ForceCredentialsMode cloudcredop.CloudCredentialsMode `json:"forceCredentialsMode,omitempty"`
+}
+```
+
+The installer will then take the value defined in the `forceCredentialsMode`
+field to populate the config for the cloud-credential-operator so that CCO knows
+what mode it should be forced into.
+
+### cloud-credential-operator
+Formalize the constants in cloud-credential-operator repo to export:
+```
+type CloudCredentialsMode string
+
+const (
+	// MintMode is the annotation on the cluster's cloud credentials to
+	// indicate that CCO should be creating users for each
+	// CredentialsRequest.
+	MintMode CloudCredentialsMode = "mint"
+
+	// PassthroughMode is the annotation on the cluster's cloud credentials
+	// to indicate that CCO should just copy over the cluster's cloud
+	// credentials for each CredentialsRequest.
+	PassthroughMode CloudCredentialsMode = "passthrough"
+)
+```
+
+Introduce a config object to allow modifying the runtime behavior of CCO.
+
+```
+type CloudCredentialOperatorConfig struct {
+	Spec CloudCredentialOperatorConfigSpec `json:"spec"`
+}
+
+type CloudCredentialOperatorConfigSpec struct {
+	// ForceCredentialsMode will instruct CCO to skip any permissions
+	// checking and assume the designated mode when reconciling
+	// CredentialsRequests.
+	// +optional
+	ForceCredentialsMode CloudCredentialsMode `json:"forceCredentialsMode,omitempty"
+}
+```
+
+### User Stories [optional]
+
+#### Story 1
+
+A user installing OpenShift in an AWS account subject to SCPs would run the
+installer with the `forceCredentialsMode` field set appropriately. This will
+generate manifests for CCO containing the CloudCredentialsConfig CR representing
+the value in the install-config.yaml.
+
+```
+./openshift-install create install-config --dir my-aws-cluster
+# edit generated install-config to add the `forceCredentialsMode: "mint"` field
+./openshift-install create cluster --dir my-aws-cluster
+```
+
+This will cause the installer to skip any pre-flight permissions checks and lay
+down the manifest for CCO to indicate that `mint` mode should be assumed:
+
+```
+apiVersion: v1
+kind: CloudCredentialOperatorConfig
+metadata:
+  name: cloud-credential-operator-config
```

Comment:
```suggestion
  name: cluster
```

since this is going to be a singleton object, it is better to specify the name that operator will respond to.